src/main/java/com/soundswapper/SoundSwapperPlugin.java

```diff
@@ -52,93 +94,255 @@ SoundSwapperConfig provideConfig(ConfigManager configManager)
+
+		if (config.soundEffects())
+		{
+			if (soundClips.containsKey(soundId))
+			{
 				event.consume();
-				playCustomSound(eventSound);
+				playCustomSound(soundClips.get(soundId));
+				return;
 			}
 		}
+
+		if (config.consumeSoundEffects() || blacklistedSounds.contains(soundId))
+		{
+			if (!whitelistedSounds.isEmpty() && whitelistedSounds.contains(soundId))
+			{
+				log.debug("whitelisted other sound effect passed: {}", soundId);
+				return;
+			}
+
+			log.debug("consumed other sound effect: {}", soundId);
+			event.consume();
+		}
 	}
 
 	@Subscribe
 	public void onAreaSoundEffectPlayed(AreaSoundEffectPlayed event)
 	{
-		if (config.areaSoundEffects()) {
-			int eventSound = event.getSoundId();
-			if (clips.containsKey(eventSound)) {
+		int soundId = event.getSoundId();
+
+		if (config.areaSoundEffects())
+		{
+			if (areaSoundClips.containsKey(soundId))
+			{
 				event.consume();
-				playCustomSound(eventSound);
+				playCustomSound(areaSoundClips.get(soundId));
+				return;
+			}
+		}
+
+		if (config.consumeAreaSounds() || blacklistedSounds.contains(soundId))
+		{
+			if (!whitelistedAreaSounds.isEmpty() && whitelistedAreaSounds.contains(soundId))
+			{
+				log.debug("whitelisted area sound effect passed: {}", soundId);
+				return;
 			}
+
+			log.debug("consumed area sound effect: {}", soundId);
+			event.consume();
 		}
 	}
 
-	private boolean tryLoadSound(String sound_name, Integer sound_id)
+	private boolean tryLoadSound(HashMap<Integer, Clip> map, String sound_name, Integer sound_id)
 	{
 		File sound_file = new File(SOUND_DIR, sound_name + ".wav");
-		if (sound_file.exists()) {
-			try {
+
+		if (sound_file.exists())
+		{
+			try
+			{
 				InputStream fileStream = new BufferedInputStream(new FileInputStream(sound_file));
 				AudioInputStream sound = AudioSystem.getAudioInputStream(fileStream);
+
 				Clip clip = AudioSystem.getClip();
 				clip.open(sound);
-				clips.put(sound_id, clip);
+				map.put(sound_id, clip);
+
 				return true;
-			} catch (UnsupportedAudioFileException | IOException | LineUnavailableException e) {
+			}
+			catch (UnsupportedAudioFileException | IOException | LineUnavailableException e)
+			{
 				log.warn("Unable to load custom sound " + sound_name, e);
 			}
 		}
 
 		return false;
 	}
 
-	private synchronized void playCustomSound(Integer sound_id)
+	private void updateClipList(HashMap<Integer, Clip> clips, String configText)
+	{
+		for (int sound : clips.keySet())
+		{
+			Clip clip = clips.get(sound);
+
+			if (clip != null)
+			{
+				if (clip.isOpen())
+				{
+					clip.close();
+				}
+			}
+		}
+
+		clips.clear();
+
+		for (String s : Text.fromCSV(configText))
+		{
+			try
+			{
+				int id = Integer.parseInt(s);
+				tryLoadSound(clips, s, id);
+			} catch (NumberFormatException e)
+			{
+				log.warn("Invalid sound ID: {}", s);
+			}
+		}
+	}
+
+	private List<Integer> getIds(String configText)
+	{
+		if (configText == null || configText.isEmpty())
+		{
+			return null;
```

Comment:
Here's the source of the nullptr exception. Return empty list instead of null.

I think this just fixes the issues haha. Will have to test some more to make sure.